src/core/filters/meta/bpf/include/meta_filter.h

```diff
@@ -168,7 +185,23 @@ bool filter_bytes(struct retis_meta_ctx *ctx)
 	return !(*sp1 - *sp2);
 }
 
-static __always_inline u64 fixup_signed(u64 val, u32 sz)
+/* Removes prefix and suffix bits that can be part of a potential
+ * previous or following bitfield and extracts the bitfield into a
+ * u64.
+ * Right shift on a signed negative numbers is implementation-defined.
+ * Both clang and gcc act on them by sign extension.
+ * This allows to preseve sign while extracting the bitfield.
+ */
+static __always_inline
+u64 fixup_bf(u64 val, bool has_sign, u16 bit_off, u16 bit_sz)
+{
+	val <<= (64 - bit_sz) - (bit_off  % 8);
```

Comment:
I don't get why the `%` above. Shouldn't we just remove the offset in bits?

Nit: would have name the function `extract_bf` for more clarity, but of course not a blocker at all, only personal taste :)

Reply:
we discussed offline the first part.
The new name sgtm.